core/src/main/java/com/linecorp/armeria/client/endpoint/PropertiesEndpointGroup.java

```diff
@@ -181,14 +248,35 @@ private static void validateDefaultPort(int defaultPort) {
                       "defaultPort: %s (expected: 1-65535)", defaultPort);
     }
 
-    private final List<Endpoint> endpoints;
+    @Nullable
+    private FileWatchRegisterKey watchRegisterKey;
 
     private PropertiesEndpointGroup(List<Endpoint> endpoints) {
-        this.endpoints = endpoints;
+        setEndpoints(endpoints);
     }
 
+    private PropertiesEndpointGroup(Path path, String endpointKeyPrefix, int defaultPort) {
+        setEndpoints(loadEndpoints(
+                path,
+                requireNonNull(endpointKeyPrefix, "endpointKeyPrefix"),
+                defaultPort));
+        watchRegisterKey = registry.register(path, () ->
+                setEndpoints(loadEndpoints(path, endpointKeyPrefix, defaultPort)));
+    }
+
+    /**
+     * Unregisters the current {@link PropertiesEndpointGroup} from the {@link FileWatcherRegistry}.
+     * Invoking this method will attempt to release any unused resources used to continuously
+     * watch properties files.
```

Comment:
Just realized that this refers to `FileWatcherRegistry` which is not a public class, exposing the implementation detail of this class. How about just removing the Javadoc entirely, given that it behaves just like a usual `Closeable`?